### PR TITLE
Babel-plugin dont handle empty location

### PR DESCRIPTION
**Pull request by restrry**
| Q                        | A  |
| ------------------------ | ---  |
| Fixed Issues?            | https://github.com/redux-saga/redux-saga/issues/1742 |
| Patch: Bug Fix?          |  bug fix  |
| Major: Breaking Change?  |  no  |
| Minor: New Feature?      |    |
| Tests Added + Pass?      | Yes |
| Any Dependency Changes?  |    |

`FunctionExpression` didn't have empty location check (when code is not part of original code, but generated by other babel-plugins) so I added

### CODE REVIEW

File: packages/babel-plugin-redux-saga/test/fixtures/preset-env/source.js [hunk 1]
```diff
@@ -5,3 +5,13 @@ function* test1() {
 function* test2() {
   yield 2
 }
+
+class Component extends React.PureComponent {
```
Comment (Andarist) [inline packages/babel-plugin-redux-saga/test/fixtures/preset-env/source.js]: is this testing the reported issue though? the reporter mentions async functions being involved (which makes them trnaspiled to generators)

isnt this code here having location? (its not inserted by babel)

Comment (restrry) [reply packages/babel-plugin-redux-saga/test/fixtures/preset-env/source.js]: by whatever reason `async/await` after being processed by `regenerator` has `path.node.generator === true`.
at the same time with `env` preset both `async/await` and `function*` are replaced by regenerator and don't have locations [demo](https://babeljs.io/repl/#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAEDCD2BbADvAdgUzQFwIzQwA9ssATGAJQzGGwDoAFAVwCcMEV0ttoBvAKGjRIATzTBoAcwzYAImGxgAFAEo-godDbZWaYQHcwASx4MWSIxAx02EeCABuGJWiYgQKjQF8NbNKQwWVXVNaGB0CB5SBTBoAF5oAG06FOwAC0s6aTkY1QBdDSFtXWhoxW9-H35QSBgOVEwcACYCYjJKalpGVnYkBu4Q4QgxCWz5RWCBUOKWPTBDE2gzCysbDDtHZ1d3TyEfIrJAycKwiKiY-KSUunTMsdyVAumZErKwCq8gA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=6.26.0&envVersion=1.6.2)

Comment (Andarist) [reply packages/babel-plugin-redux-saga/test/fixtures/preset-env/source.js]: hm, this seems terrible for our use case as preset-env is really popular (as it should be - it's primary use case for babel now)

I'll try look into this as soon as possible to see how babel handle this and if there is anything we can do to overcome this

Comment (restrry) [reply packages/babel-plugin-redux-saga/test/fixtures/preset-env/source.js]: @Andarist updated my comment. to test the case reference the original issue

**Comment by szb512** [comment]
I suggest that you go here to find out how to upgrade to the latest version of babel:
https://babeljs.io/docs/en/v7-migration

**Comment by szb512** [comment]
Maybe then you can try again.

**Comment by restrry** [comment]
@szb512 actually we aim to support both babel v6 and v7. later we will remove v6 when most projects drop its support

**Pull request merged**